lib/oidcstrategy.js

```diff
@@ -210,21 +210,27 @@ Strategy.prototype.authenticate = function authenticateStrategy(req, options) {
       },
       // handle error response
       (next) => {
-        if (!(req.query && req.query.code)) {
+        var err = undefined;
+        var err_description = undefined;
+
+        if (req.query && req.query.error) {
+          err = req.query.error;
+          err_description = req.query.error_description;
+        } else if (req.body && req.body.error) {
+          err = req.body.error;
+          err_description = req.body.error_description;
+        } else {
           return next();
         }
-        // @TODO: need more specific error handling
-        //        response might include error_code, error_details and more
 
-        // Error information pertaining to OAuth 2.0 flows is encoded in the
-        // query parameters, and should be propagated to the application.
+        log.info('Error received was: ', err);
+        if (err_description)
+          log.info('Error description was: ', err_description);
 
-        log.info(`OPENID CONNECT ERROR: We are in the OpenID Connect Response with error.
-          Assumed because no auth code was in the query.`);
-        log.info('Error received was: ', req.query.error);
-
-        // pass error to `next` and handle error occurance centrally in `async.waterfall`'s callback
-        return next(req.query.error);
+        // Unfortunately, we cannot return the 'error description' to the user, since 
+        // it goes to http header by default and it usually contains characters that
+        // http header doesn't like, which causes the program to crash. 
```

Comment:
Should we be URL encoding the description so it can be in the header without crashing? Is it possible clients are expecting the error description in the header to be shown to the user?